### PR TITLE
Bump k8s.io/client-go to 1.28.2

### PR DESCRIPTION
**Pull request by MartinForReal**
bump k8s.io/client-go to 1.28.2 and resolve issues in go.mod

### CODE REVIEW

**Review by hakman — CHANGES_REQUESTED** [review]

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    directory: "/"
+    open-pull-requests-limit: 1
+    schedule:
+      interval: daily
+      time: "01:00"
+    labels:
+      - "area/dependency"
+      - "ok-to-test"
+      - "lgtm"
+      - "approved"
```
[the remark this answers is not in the document]
Comment (MartinForReal) [reply .github/dependabot.yml:13]: yes. The dependencies will be updated if the pr passed e2e tests. And pr will be blocked if the pr is modified. (tide will remove lgtm label). In that case, reviewer need to review the pr.

[the remark this answers is not in the document]
Comment (hakman) [reply .github/dependabot.yml:13]: Considering the dependency on aws-sdk-go, I remember seeing endless dependabot PRs for its frequent updates.

Another option is the kOps approach, a weekly job that updates the direct dependencies, which is much less noisy with many dependencies:
https://github.com/kubernetes/kops/pull/15890
https://github.com/kubernetes/kops/blob/master/.github/workflows/update-deps.yml

[the remark this answers is not in the document]
Comment (MartinForReal) [reply .github/dependabot.yml:13]: I can change it to update by weekly

[the remark this answers is not in the document]
Comment (hakman) [reply .github/dependabot.yml:13]: I would suggest to separate the bumping of dependencies to a new PR and continue the discussion here on how and if automated dependency updates should look like.

At the moment I see these 2 options:
* dependabot that runs once a week with 5-10 open PRs at most
* script that runs once a week and does all the updates in 1 PR, but with less info on what changed

@vteratipally @mmiranda96 Any thoughts?

[the remark this answers is not in the document]
Comment (MartinForReal) [reply .github/dependabot.yml:13]: opened a pr for dependabot config https://github.com/kubernetes/node-problem-detector/pull/811

**Comment by MartinForReal** [comment]
@hakman @vteratipally Could you please take a look? Thanks!

File: test/go.mod [hunk 2]
```diff
@@ -6,36 +6,37 @@ replace k8s.io/node-problem-detector => ../.
...
 	github.com/spf13/pflag v1.0.5
-	golang.org/x/crypto v0.11.0
-	golang.org/x/oauth2 v0.9.0
-	google.golang.org/api v0.114.0
-	k8s.io/apimachinery v0.28.1
+	golang.org/x/crypto v0.13.0
+	golang.org/x/oauth2 v0.12.0
+	google.golang.org/api v0.140.0
+	k8s.io/apimachinery v0.28.2
 	k8s.io/component-base v0.28.1
```
Comment (hakman) [inline test/go.mod:18]:
```suggestion
	k8s.io/component-base v0.28.2
```

**Review by hakman — CHANGES_REQUESTED** [review]

**Comment by MartinForReal** [comment]
<html>
<body>


/test pull-npd-e2e-node
--


<br class="Apple-interchange-newline">
</body>
</html>

**Review by hakman — APPROVED** [review]

**Pull request merged**